subsys/net/openthread/report/CMakeLists.txt

```diff
@@ -0,0 +1,60 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+include(report_generation_utils.cmake)
+include(${ZEPHYR_NRFXLIB_MODULE_DIR}/openthread/cmake/extensions.cmake)
+
+ot_report_add_message("################### OPENTHREAD REPORT ###################" TRUE)
+
+# Add basic information
+ot_report_add_message("+ Target device: ${CONFIG_SOC}" TRUE)
+ot_report_add_message("+ Thread version: v${CONFIG_OPENTHREAD_THREAD_VERSION}" TRUE)
+
+# Add feature set
+if(CONFIG_OPENTHREAD_NORDIC_LIBRARY_MASTER)
+    set(feature_set "master")
+elseif(CONFIG_OPENTHREAD_NORDIC_LIBRARY_FTD)
+    set(feature_set "ftd")
+elseif(CONFIG_OPENTHREAD_NORDIC_LIBRARY_MTD)
+    set(feature_set "mtd")
+elseif(CONFIG_OPENTHREAD_NORDIC_LIBRARY_RCP)
+    set(feature_set "rcp")
+else()
+    set(feature_set "custom")
+endif()
+ot_report_add_message("+ Feature set: ${feature_set}" TRUE)
+
+# Check whether OpenThread is from library or built from sources
+if(CONFIG_OPENTHREAD_SOURCES)
+    ot_report_add_message("+ OpenThread library has been built from sources" TRUE)
+else()
+    openthread_calculate_lib_path("v${CONFIG_OPENTHREAD_THREAD_VERSION}" LIB_PATH)
+    cmake_path(RELATIVE_PATH LIB_PATH BASE_DIRECTORY ${ZEPHYR_NRFXLIB_MODULE_DIR} OUTPUT_VARIABLE LIB_PATH_REL)
+    ot_report_add_message("+ OpenThread Library from nrfxlib: ${LIB_PATH_REL}" TRUE)
+endif()
+
+# Get OpenThread revision
+set(OPENTHREAD_VERSION "" CACHE STRING "OpenThread revision")
+if(OPENTHREAD_VERSION STREQUAL "")
+    ot_report_git_version(OPENTHREAD_VERSION ${ZEPHYR_OPENTHREAD_MODULE_DIR})
+    ot_report_add_message("+ OpenThread NCS revision: ${OPENTHREAD_VERSION}" TRUE)
+endif()
+
+# Get NCS revision
+set(NCS_VERSION "" CACHE STRING "NCS revision")
+if(NCS_VERSION STREQUAL "")
+    ot_report_git_version(NCS_VERSION ${ZEPHYR_NRF_MODULE_DIR})
+    ot_report_add_message("+ NCS revision: ${NCS_VERSION}" TRUE)
+endif()
+
+# Diff of nrfxlib directory between the current and the latest NCS release revisions.
+string(SUBSTRING ${NCS_VERSION} 0 6 LATEST_NCS_RELEASE)
+set(NRFXLIB_DIFF "" CACHE STRING "nrfxlib diff to the recent revision")
+ot_report_git_diff(NRFXLIB_DIFF ${LATEST_NCS_RELEASE} ${ZEPHYR_NRFXLIB_MODULE_DIR} "openthread")
+ot_report_add_message("+ Changes between the current NCS revision (${NCS_VERSION}) and the recent NCS release (${LATEST_NCS_RELEASE}) in the nrfxlib directory\n" FALSE)
+ot_report_add_message(${NRFXLIB_DIFF} FALSE)
```

Comment:
If there are differences in the nrfxlib directory found, could we add stdout message to warn that they exist and full diff can be found in generated report file?

Reply:
ok, sounds good, I will add it :)